src/operators/products.c

```diff
@@ -0,0 +1,221 @@
+/*
+   IGraph library.
+   Copyright (C) 2025  The igraph development team <igraph@igraph.org>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "igraph_constructors.h"
+#include "igraph_operators.h"
+
+#include "igraph_conversion.h"
+#include "igraph_interface.h"
+
+#include "core/interruption.h"
+#include "math/safe_intop.h"
```

Comment:
The convention we use for header includes is:

 1. First comes the header where the functions defined in this file are declared.
 2. This is followed by a newline, then public headers alphabeticaly.
 3. A newline, then private headers alphabetically.
 4. Finally, any standard library headers comes last, though these are rarely needed.

Can you please reorder these and remove unused headers? A good IDE helps you determine what is unused, although sometimes they give incorrect information. I believe `#include "igraph_conversion.h"` is not needed here.